python/tank/folder/configuration.py

```diff
@@ -221,8 +221,14 @@ def _load_schema(self, schema_config_path):
             if metadata is None:
                 if os.path.basename(project_folder) == "project":
                     # this is a project folder with no project.yml file specified
-                    # in this case, just assume it is the primary storage 
-                    metadata = {"type": "project", "root_name": constants.PRIMARY_STORAGE_NAME}
+                    # in this case, check if we have a single storage and use it
+                    local_roots = self._tk.pipeline_configuration.get_local_storage_roots()
+                    if len(local_roots) == 1:
+                        metadata = {"type": "project", "root_name": local_roots.keys()[0]}
+                    else:
+                        # Fallback to "primary" storage if we don't have any root
```

Comment:
i think this should be an `elif len(local_roots) > 1:` to indicate that it's now only for the multi-root case that we require a primary. I suggest tweaking the comment to not say fallback, that confused me. The logic doesn't have fallbacks as far as i understand it - if you have a single storage, then that will be the 'main' one. If you have more than one storage, the primary one becomes the 'main' one. If you have zero storage roots, an exception should be raised.